zulipterminal/ui_tools/buttons.py

```diff
@@ -29,8 +29,14 @@ def __init__(self, controller: Any, caption: str,
         super().__init__("")
         self._w = self.widget(count)
         self.controller = controller
+        self.set_muted_streams()
         urwid.connect_signal(self, 'click', self.activate)
 
+    def set_muted_streams(self) -> None:
+        if self.caption in [self.controller.model.stream_dict[ele]['name']
+                            for ele in self.controller.model.muted_streams]:
+            self.update_count(-1)
```

Comment:
The reason I put this as a method was so that I could patch it easily through tests. `update_count (-1)` seems to the only way we can set a stream to be muted. Do you have anything in mind which avoids this?

Reply:
This works as a way to simplify the tests, I agree, though we could place the model access elsewhere if we refactor later, which may make this simpler. For now, let's get it working :)

Reply:
Right. Sure that makes sense :)